front/cypress/e2e/admin/pages_and_menu/update_homepage_content.cy.ts

```diff
@@ -0,0 +1,94 @@
+import { randomString } from '../../../support/commands';
+
+describe('Admin: update HomePage content', () => {
+  before(() => {
+    cy.setAdminLoginCookie();
+    cy.visit('/admin/pages-menu/');
+    cy.acceptCookies();
+  });
+
+  it('updates top and bottom info section content and visibility', () => {
+    cy.intercept('PATCH', '**/home_page').as('saveHomePage');
+    cy.intercept('GET', '**/home_page').as('getHomePage');
+
+    const topInfoContent = randomString();
+    const bottomInfoContent = randomString();
+
+    cy.apiUpdateHomepageSettings({
+      top_info_section_enabled: false,
+      bottom_info_section_enabled: false,
+      events_widget_enabled: false,
+    });
+    // go to page with homepage settings toggles
+    cy.get('[data-testid="edit-button"]').first().click();
+
+    // visit top into section edit page
+    cy.get('[data-cy="e2e-admin-edit-button"]').eq(1).click();
+    cy.get('#custom-section-en').clear().type(topInfoContent);
+    cy.get('.e2e-submit-wrapper-button').click();
+
+    // // go back home
+    cy.get('[data-cy="breadcrumbs-Home"]').click();
+
+    // // visit bottom into section edit page
+    cy.get('[data-cy="e2e-admin-edit-button"]').eq(2).click();
+    cy.get('#custom-section-en').clear().type(bottomInfoContent);
+    cy.get('.e2e-submit-wrapper-button').click();
+
+    cy.visit('/');
+
+    cy.get('[data-testid="e2e-landing-page-top-info-section"]').should(
+      'not.exist'
+    );
+    cy.get('[data-testid="e2e-landing-page-bottom-info-section"]').should(
+      'not.exist'
+    );
+    cy.get('[data-testid="e2e-events-widget-container"]').should('not.exist');
+
+    cy.visit('/admin/pages-menu/');
+    cy.get('[data-testid="edit-button"]').first().click();
+
+    // toggle top info section and wait for requests to complete
+    cy.get(
+      '[data-cy="e2e-admin-section-toggle-top_info_section_enabled"]'
+    ).click();
+    cy.wait('@saveHomePage');
+    cy.wait('@getHomePage');
+
+    // wait for the next toggle to become disabled, then enabled again once it finishes toggling
+    cy.get('[data-cy="e2e-admin-section-toggle-bottom_info_section_enabled"]')
```

Comment:
This is kind of a hack to bypass the `wait(1000)` I had earlier, to wait for the toggle to become disabled (request in progress) and then enabled again (request done). I don't think just waiting for the request works because the toggle has animation on it, so it will still be disabled once the request has finished successfully. Not the prettiest thing but it seems to work.

Reply:
It's a bit strange - perhaps we could think of a better loading state than disabling the toggles in the future 🤔 but it's ok if this is what you settled on for now.

Reply:
The initial idea was to avoid weird race conditions or needing complex state management locally, since the request to enable/disable is sent immediately after the user presses the toggle, and all three toggles use the same endpoint/response. Disabling the toggles means they're only ever enabled and "toggled" according to the actual state of the backend response, so I figured the accurate feedback was worth it. If we have a better pattern for this lmk :)